Assets/js/resultspage.js

```diff
@@ -0,0 +1,169 @@
+$(document).ready(function () {
+  function yelpRequest(zipCode) {
+    foodRequest(zipCode);
+    eventRequest(zipCode);
+    $(".modal").modal();
+  }
+  // Gets the zip code from the URL
+  const urlParams = new URLSearchParams(window.location.search);
+  const zipCode = urlParams.get("zip");
+  yelpRequest(zipCode);
+});
+
+// Function that gets Yelp API request & builds UI content
+function foodRequest(zipCode) {
+  let yelpAPIKey =
+    "nfTUVrFeoPDHChVBIUtbNHnP1lRBrqWO50jNjHHoePVDf4Q2IVoCGIld9YgAq-MX9VDeGBNUb0plvpnIFdfuf7X3huELvNrBDN2YjSEKDf3ANCXhx3-gJb9JnvL9X3Yx";
+
+  // Forced to go through herokuapp proxy since we do not have server setup
+  let yelpREST = axios.create({
+    baseURL: "https://cors-anywhere.herokuapp.com/https://api.yelp.com/v3",
+    headers: {
+      Authorization: `Bearer ${yelpAPIKey}`,
+      "Content-type": "application/json",
+    },
+  });
+
+  // Function that builds url for Yelp API request & UI content
+  yelpREST("/businesses/search", { params: { location: zipCode } }).then(
+    ({ data }) => {
+      // Get a random number between 0 and legnth of array
+      let randomRestIndex = Math.floor(Math.random() * data.businesses.length);
+      let restData = data.businesses[randomRestIndex];
+
+      // Creating <img> element and adding image url to src attr.
+      let foodImage = $("<img>").attr({
+        src: restData?.image_url,
+        height: "auto",
+        width: "100%",
+        fit: "scale",
+      });
+      foodImage.remove();
+
+      // Puts full address together
+      let fullRestAddress = restData.location.display_address.join(" | ");
+
+      // Using IF statements to ensure data is available, if not, dont display
+      // Restaurant image
+      if (restData.image_url !== undefined || "") {
+        $("#restImage").empty();
+        $("#restImage").append(foodImage);
+      }
+      // Restaurant name
+      $("#restName").text(`Name: ${restData.name}`);
+      // Restaurant type
+      if (restData.categories[0].title !== undefined || "") {
+        $("#restType").text(`Type: ${restData.categories[0]?.title}`);
```

Comment:
You want to be really careful with using the latest greatest features on the front end without running them through babel.

You might not have a ton of browser support.  On the backend, you usually can do whatever you want.  On the front-end you want to try to be a little more vanilla so that your website doensn't break on older or out of date browsers.

You can always check this page for level of support

https://caniuse.com/?search=optional%20chaining